base64.js

```diff
@@ -1,7 +1,7 @@
 ;(function () {
 
   var object =
-    typeof exports != 'undefined' ? exports :
+    typeof exports === 'object' && !!exports && !exports.nodeType ? exports : // null and ID
```

Comment:
Is `!!exports` just to handle `null`? If so, let's use `exports !== null` instead to avoid type coercion.

Could you explain `!exports.nodeType`?

Reply:
> Could you explain !exports.nodeType?

https://www.w3.org/Bugs/Public/show_bug.cgi?id=11960

> let's use exports !== null instead

The problem is that, in browsers, there's another object which can be falsy: `document.all`. The type coercion takes care of that case too.

Reply:
I'm sorry for dropping this pull request, @Mouvedia.

> > Is `!!exports` just to handle `null`? If so, let's use `exports !== null` instead to avoid type coercion.
>
> The problem is that, in browsers, there's another object which can be falsy: `document.all`.

This is new to me. You are quite right:

```javascript
> Boolean (document.all)
false
```

That said, we won't evaluate the second `&&` operand if `exports` is `document.all`:

```javascript
> typeof document.all === 'object'
false

> typeof document.all
'undefined'
```

Would the following predicate address your concerns?

```javascript
typeof exports === 'object' && exports !== null && exports.nodeType !== 1
```

Reply:
> `exports.nodeType !== 1`

This is incorrect. If you don't want `!exports.nodeType` you will have to expand it to `typeof exports.nodeType === 'undefined'`. `!exports.nodeType` works because nodeType 0 doesn't exist.

cf https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

> we won't evaluate the second `&&` operand if `exports` is `document.all`

Not exactly. I was being overly cautious just in case a browser did one (typeof 'object') but not the other (falsy). It's probably not necessary in hindsight.
It seems you want to be more explicit, if so replace

```
typeof exports === 'object' && !!exports && !exports.nodeType
```
with
```
typeof exports === 'object' && exports !== null && typeof exports.nodeType === 'undefined'
```

Iv given you the rights to edit.

Reply:
> ```
> typeof exports === 'object' && exports !== null && typeof exports.nodeType !== 'number'
> ```

This works for me. :thumbsup:

I think `exports.nodeType !== 1` would also work, because only an `Element` node can have an `id` attribute. I may be wrong, though, and the `typeof` check seems more elegant regardless.

Would you like to update the pull request or would you like me to do so?

Reply:
> because only an `Element` node can have an `id` attribute

Probably, yes.

> would you like me to do so?

Please do.